pages/index.tsx

```diff
@@ -1,30 +1,126 @@
+import { ContentBlockAdapter } from '../components/presenters/ContentPanel/adapters/ContentBlockAdapter'
+import { LiveExampleAdapter } from '../components/presenters/LiveExample/adapters/LiveExampleAdapter'
+import { CodeBlockAdapter } from '../components/presenters/CodeBlock/adapters/CodeBlockAdapter'
+import { ApiTableAdapter } from '../components/presenters/ApiTable/adapters/ApiTableAdapter'
 
-export default function Home({ posts }) {
-  return (
-    <LayoutFramework>
-      {posts.map(({ title }: PostType) => {
-        return <Post title={title} />
-      })}
-    </LayoutFramework>
-  )
+export type ContentType =
+  | 'Hero'
+  | 'ContentBlock'
+  | 'LiveExample'
+  | 'CodeBlock'
+  | 'ApiTable'
+
+interface HomeProps {
+  sections: any
 }
 
 export async function getStaticProps() {
-  const res = await fetchEntries()
-
-  const posts = await res.map((post: any) => {
-    return post.fields
-  })
+  const response = await fetchContent(`
+    {
+      sectionCollection(limit: 5) {
+        items {
+          title
+          articleCollection(limit: 10) {
+            items {
+              ... on Article {
+                contentCollection(limit: 10) {
+                  items {
+                    __typename
+                    ... on Hero {
+                      title
+                      heroDescription: description
+                    }
+                    ... on ContentBlock {
+                      title
+                      description {
+                        json
+                      }
+                    }
+                    ... on LiveExample {
+                      title
+                      description {
+                        json
+                      }
+                    }
+                    ... on CodeBlock {
+                      title
+                      description {
+                        json
+                      }
+                      filename
+                      sourceCode
+                    }
+                    ... on ApiTable {
+                      title
+                      apiTableDescription: description
+                      apiFieldCollection(limit: 10) {
+                        items {
+                          ... on ApiField {
+                            name
+                            dataType
+                            defaultValue
+                            description
+                            required
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `)
 
   return {
     props: {
-      posts,
+      sections: response?.sectionCollection?.items ?? [],
     },
   }
 }
+
+/**
+ * Map content types to components using adapters
+ *
+ */
+function renderPresenter(type: ContentType, fields: any): React.ReactElement {
+  const componentMap = {
```

Comment:
Maybe there is a prettier way to do this? To avoid the repetition with the fields prop?

Reply:
I think there probably is as was the observation on #71